pkgs/applications/office/paperless-ngx/default.nix

```diff
@@ -74,6 +70,18 @@ let
           hash = "sha256-KWkMV4L7bA2Eo/u4GGif6lmDNrZAzvYyDiyzyWt9LeI=";
         };
       });
+
+      # Patched version of the library with django 4 compatibility
+      # upstreams pipfile.lock uses the 0.3.0 version, but with that, the build / test phase fails in NixOS
+      djangorestframework-guardian = super.djangorestframework-guardian.overridePythonAttrs (oa: rec {
+        version = "0.5.0";
+        src = fetchFromGitHub {
+          owner = "johnthagen";
+          repo = "django-rest-framework-guardian2";
+          rev = "v${version}";
+          hash = "sha256-aW20xEmVTAgwayWMJsabmyKNW65NftJyQANtT6JV74U=";
+        };
+      });
```

Comment:
Properly package that up as `djangorestframework-guardian2`.

https://github.com/johnthagen/django-rest-framework-guardian2/blob/master/setup.cfg#L24